### ISSUE TITLE
Reassigned Assignments leads to incorrect Sandbox Urls

### ISSUE THREAD

**Issue by bwreid** [WikiEducationFoundation/WikiEduDashboard#3434]
### What is happening?

It is possible to create a condition where the `sandbox_url` for an assignment is incorrectly pointing towards the wrong person's sandbox. This will only happen if multiple people are assigned to an article or someone is reviewing an article.

### To Reproduce

* Article with title of `AssignmentA` is given to user with a username of "UsernameX"
* `AssignmentA` will have a `sandbox_url` with something like `/User:UsernameX/AssignmentA`
* Another user is given a review assignment (`ReviewB`) that reviews `AssignmentA`. The `sandbox_url` will be assigned to the existing `sandbox_url`
* `AssignmentA` is removed from "UsernameX" and assigned instead to "UsernameY"
* `ReviewB` still points to the "UsernameX" sandbox, not "UsernameY"

### Expected behavior

The `sandbox_url` should be updated on reassignment. This could likely be done before an assignment is being deleted.

Additionally, if possible we should check to see if revisions already exist on the prior sandbox. If they do, we should not change the `sandbox_url`

**Comment by shishiro26** [comment]
yupp got it

**Comment by shishiro26** [comment]
@ragesoss, I have been working on this issue, but I haven’t been able to achieve the desired result. Could you provide some suggestions on how to resolve this?

**Comment by ragesoss** [comment]
@shishiro26 can you share what you've tried so far, or any details about how you're stuck?

**Comment by shishiro26** [comment]
In the `assignment.rb` file, the `sandbox_url` is being set. I want to modify this logic to check if a user has made any edits to the sandbox using a helper method that counts the characters edited. If the character count is greater than zero, the `sandbox_url` should remain with the first user who edited it. If no edits have been made, then the sanboxurl will be updated to the next user who is being assigned . However, when I edit the sandbox in my local development environment, I don’t see any revisions reflected in my MySQL database

**Comment by ragesoss** [comment]
The Dashboard currently doesn't automatically check the contents of a wiki sandbox for an assignment, so it will probably require writing code to do that.

**Comment by shishiro26** [comment]
Yes, this won't be useful for the new courses and I just need to check the course type and hide the button accordingly, right?

**Comment by shishiro26** [comment]
How can I load the revisions made in the sandbox into my local MySQL database? I don’t see any revisions from my sandbox draft in the MySQL server. Wouldn’t it be useful to check the number of characters added in the revisions in assigning the sandbox_url? .By directly calling the Wiki API in the `assigment.rb` file, checking the number of characters added, and assigning the `sandbox_url` accordingly.
